lib/compilation_engine.rb

```diff
@@ -129,5 +280,14 @@ def output_token
     end
 
     @output.puts("<#{token_type}> #{CGI.escapeHTML(token)} </#{token_type}>")
+    advance
+  end
+
+  def is_symbol_token_and_equal?(symbol)
+    @tokenizer.token_type == :SYMBOL && @tokenizer.symbol == symbol
+  end
+
+  def is_keyword_token_and_equal?(keyword)
```

Comment:
If the `keyword` argument was optional, or a `*keywords` splat of multiple alternatives, we could use this helper in `#compile_statements` as suggested above.